### PR TITLE
Integrate gutenberg-mobile release 1.55.2

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description
This PR incorporates the 1.55.2 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3650

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * [*] Block Editor: "Set as featured" button within image block settings. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3116]
 * [***] Block Editor: Audio block now available on WP.com sites on the free plan. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3523]
 * [*] Block Editor: Improve unsupported block message for reusable block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3621]
+* [**] Fix incorrect block insertion point after blurring the post title field. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3640]
```
Comment (fluiddot) [inline RELEASE-NOTES.txt]: I see the rest of the entry items related to Gutenberg have the prefix `Block Editor:`, I'm not sure if it's a convention but we might want to include it, wdyt?

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**[Writing flow](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3650#issuecomment-866177314) and [Unsupported block](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3650#issuecomment-866669414) tests passed 🟢 .**

_I only added a comment regarding the release notes but it's just a minor issue._

**Comment by fluiddot** [comment]
I saw that the optional tests (`ci/circleci: Optional Tests/Hold` CI job) weren't triggered so I went ahead and run them:
https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/24834/workflows/1152dd66-78c3-4cd4-a510-b87a9023293c/jobs/110951

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (AliSoftware) [reply RELEASE-NOTES.txt]: The release notes for 17.6 have already been sent to Editorial on first week of the Sprint, and already sent to our translation vendor last week too, so those won't make it to the PlayStore anyway (as [the Peril warning above](https://github.com/wordpress-mobile/WordPress-Android/pull/14892#issuecomment-866092312) explains).

Comment (dcalhoun) [reply RELEASE-NOTES.txt]: Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/14892/commits/e24729b57deac65dab8aae1b6af420b62716b36b.

**Comment by AliSoftware** [comment]
This finally landed in the new beta 17.6-rc-4 🎉

**Pull request merged**